modules/optimeraRtdProvider.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * This module adds optimera provider to the real time data module
+ * The {@link module:modules/realTimeData} module is required
+ *
+ * The module will fetch targeting values from the Optimera server
+ * and apply the tageting to each ad request. These values are created
+ * from the Optimera Mesaurement script which is installed on the
+ * Publisher's site.
+ *
+ * @module modules/optimeraRtdProvider
+ * @requires module:modules/realTimeData
+ */
+
+/**
+ * @typedef {Object} ModuleParams
+ * @property {string} clientID
+ * @property {string} optimeraKeyName
+ * @property {string} device
+ */
+
+import * as utils from '../src/utils.js';
+import { submodule } from '../src/hook.js';
+import { ajaxBuilder } from '../src/ajax.js';
+
+/** @type {ModuleParams} */
+let _moduleParams = {};
+
+/**
+ * Default Optimera Key Name
+ * This can default to hb_deal_optimera for publishers
+ * who used the previous Optimera Bidder Adapter.
+ * @type {string} */
+export let optimeraKeyName = 'hb_deal_optimera';
+
+/**
+ * Optimera Score File Base URL.
+ * @type {string}
+ */
+export const scoresBaseURL = 'https://dyv1bugovvq1g.cloudfront.net/';
+
+/**
+ * Optimera Score File URL.
+ * @type {string}
+ */
+export let scoresURL;
+
+/**
+ * Optimera Client ID.
+ * @type {string}
+ */
+export let clientID;
+
+/**
+ * Optional device parameter.
+ * @type {string}
+ */
+export let device = 'default';
+
+/**
+ * Targeting object for all ad positions.
+ * @type {string}
+ */
+export let optimeraTargeting = {};
+
+/**
+ * Make the request for the Score File.
+ */
+export function scoreFileRequest() {
+  utils.logInfo('Fetch Optimera score file.');
+  const ajax = ajaxBuilder();
+  ajax(scoresURL,
```

Comment:
why not use https://github.com/prebid/Prebid.js/blob/804295aa2dae0484d67891b73fcfc401ef8244f1/src/adloader.js#L21